Jenkinsfile

```diff
@@ -0,0 +1,149 @@
+        'gcc.release.unity',
+        'coverage'] as String[]
+
+    // create a map of all builds
+    // that we want to run. The map
+    // is string keys and node{} object values
+    def builds = [:]
+    for (int index = 0; index < variants.size(); index++) {
+        def bldtype = variants[index]
+        builds[bldtype] = {
+            node('rippled-dev') {
+                checkout scm
+                dir ('build') {
+                    deleteDir()
+                }
+                def cdir = upDir(pwd())
+                echo "BASEDIR: ${cdir}"
+                def compiler = getCompiler(bldtype)
+                def target = getTarget(bldtype)
+                if (compiler == "coverage") {
+                    compiler = 'gcc'
+                }
+                echo "COMPILER: ${compiler}"
+                echo "TARGET: ${target}"
+                def clang_cc  = (compiler == "clang") ? '/opt/llvm-4.0.0/bin/clang' : ''
+                def clang_cxx = (compiler == "clang") ? '/opt/llvm-4.0.0/bin/clang++' : ''
+                def ucc = isNoUnity(target) ? 'true' : 'false'
+                echo "USE_CC: ${ucc}"
+                withEnv(["CCACHE_BASEDIR=${cdir}",
+                         'LCOV_ROOT=""',
+                         "TARGET=${target}",
+                         "CC=${compiler}",
+                         'BUILD=cmake',
+                         'VERBOSE_BUILD=true',
+                         "CLANG_CC=${clang_cc}",
+                         "CLANG_CXX=${clang_cxx}",
+                         "CCACHE_LOGFILE=${bldtype}.ccache.txt",
+                         "USE_CCACHE=${ucc}"])
+                {
+                    myStage(bldtype)
+                    try {
+                        if (fileExists("${bldtype}.ccache.txt")) {
+                            sh "rm -f ${bldtype}.ccache.txt"
+                        }
+                        sh "ccache -s > ${bldtype}.txt"
+                        // the devtoolset from SCL gives us a recent gcc. It's not strictly needed
+                        // when we are building with clang, but it doesn't seem to interfere either
+                        sh "source /opt/rh/devtoolset-6/enable && (/usr/bin/time -p ./bin/ci/ubuntu/build-and-test.sh 2>&1) 2>&1 >> ${bldtype}.txt"
+                        sh "ccache -s >> ${bldtype}.txt"
+                    }
+                    catch (any) {
+                        throw any
+                    }
+                    finally {
+                        def outstr = readFile("${bldtype}.txt")
+                        def st = getResults(outstr)
+                        def time = getTime(outstr)
+                        def txtcolor = getFailures(outstr) == 0 ? "DarkGreen" : "Crimson"
+                        outstr = null
+                        def shortbld = bldtype
+                        shortbld = shortbld.replace('debug', 'dbg')
+                        shortbld = shortbld.replace('release', 'rel')
```

Comment:
I see what you mean. I like the shorter version now that I see the difference. I'm going to suggest a few other changes to make the thing even shorter:

- Get the number of suites or cases instead of the number of tests (`361225 tests` vs `154 suites` vs `960 cases`)
- `shortbld.replace('failures', 'failed')
- `shortbld.replace('unity', 'un')